nautobot/utilities/testing/views.py

```diff
@@ -799,6 +795,30 @@ def test_list_view_plugin_banner(self):
                 f"<div>You are viewing a table of {self.model._meta.verbose_name_plural}</div>", response_body
             )
 
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+        def test_queryset_to_csv(self):
+            # Built-in CSV export
+            if hasattr(self.model, "csv_headers"):
+                response = self.client.get(f"{self._get_url('list')}?export")
+                self.assertHttpStatus(response, 200)
+                self.assertEqual(response.get("Content-Type"), "text/csv")
+                instance1 = self._get_queryset().first()
+                # With filtering
+                response = self.client.get(f"{self._get_url('list')}?export&id={instance1.pk}")
+                self.assertHttpStatus(response, 200)
+                self.assertEqual(response.get("Content-Type"), "text/csv")
+                response_body = response.content.decode(response.charset)
+                reader = csv.DictReader(StringIO(response_body))
+                data = [dict(row) for row in reader]
+                csv_headers = getattr(self.model, "csv_headers")
+                for _, entry in enumerate(data):
+                    # x might be None, in that case use an empty sting
+                    instance1_csv_data = [str(x) if x is not None else "" for x in instance1.to_csv()]
+                    for header in csv_headers:
+                        # ignore "", None and etc, we only want "name", "slug" or similar identifiers
+                        if entry[header]:
+                            self.assertIn(entry[header], instance1_csv_data)
```

Comment:
```suggestion
                data = [dict(row) for row in reader][0]
                instance1_csv_data = [str(x) if x is not None else "" for x in instance1.to_csv()]
                # append instance
                instance1_ct_values = list(instance1.get_custom_fields().values())
                instance1_csv_data += instance1_ct_values
                # Since values in `data` are all in str; cast all values in instance1_csv_data to str
                instance1_csv_data = [str(val) for val in instance1_csv_data]
                self.assertEquals(instance1_csv_data, list(data.values()))
```
Since this approach takes into account customs types, I think this might be good to consider. I also don't see the need to obtain the csv headers, unless you want to add a test to determine whether the `instance` csv headers matches the response csv header, which I think would also be a reasonable thing to test for.

Reply:
Thank you!